### ISSUE TITLE
Redirects to website

### ISSUE THREAD

**Issue by ghost** [FreshRSS/FreshRSS#3095]
Hello,

I know it's a weird request but is it possible when I clic on a title post on the homepage it redirects to the "original" website and not open the embed view ? 

Thank you !!

**Comment by Alkarex** [comment]
Hello,
Not exactly what you ask, but this is what the little arrow is doing 
![image](https://user-images.githubusercontent.com/1008324/86522359-de0ec500-be5c-11ea-94a2-46dbf4726c80.png)

And when using keyboard navigation, this is what the <kbd>space</kbd> shortcut does as well

**Comment by ghost** [comment]
Hello, thank you for your suggestion but I prefer to read articles on original website. I'll continue to search a solution ^^

**Comment by Alkarex** [comment]
> I prefer to read articles on original website. I'll continue to search a solution ^^

This is what the little arrow icon is doing, as well as the <kbd>space</kbd> keyboard shortcut ;-)

**Comment by ghost** [comment]
Okay, I know, I am stubborn. I installed the extension CustomJS with this JS to open all links in a new tab :
`document.querySelectorAll("a").forEach(a => a.target = '_blank' );`

It works on all links... except news. If you can help me :(
Thank you =)

**Comment by Frenzie** [comment]
What do you mean by news?

**Comment by ghost** [comment]
This : 
![Capture d’écran 2020-10-17 203937](https://user-images.githubusercontent.com/206181/96350984-ef50aa00-10b8-11eb-99ea-911c68b3c8ff.jpg)

**Comment by Frenzie** [comment]
The titles? They have a few event listeners that interfere with standard behavior.

https://github.com/FreshRSS/FreshRSS/blob/9ab05c7f67a04409245b97c7c22a746b32e07847/p/scripts/main.js#L881-L1038

I'm not immediately sure what the best way to override that would be, and I don't have time to experiment today/the next two weeks since I'm in the middle of moving.

One example could be something along these lines, but that might do more than you bargained for.

```js
stream.addEventListener('click', function (ev) { ev.stopPropagation(); }, true);
```

**Comment by Alkarex** [comment]
I think what @Galiwyx means is new articles. There is an event `freshrss:load-more`, which you can register to, in order to react when new articles are loaded

**Comment by ghost** [comment]
No sorry I mean all articles, I didn't draw all arrows :D

Okay @Frenzie , I'll check that.

**Comment by KatyPurry** [comment]
> ```js
> stream.addEventListener('click', function (ev) { ev.stopPropagation(); }, true);
> ```

Hey, I am also interested in this modification. I tested but it works sporadically.

**Comment by math-GH** [comment]
If you use a mouse device with a scroll wheel/middle mouse button, just click with it on the title. It will open the article in a new tab on the original website.
Alternative: `Strg` + Mouse click on the title. It will do the same.

**Comment by TiTi** [comment]
> If you use a mouse device with a scroll wheel/middle mouse button, just click with it on the title. It will open the article in a new tab on the original website. Alternative: `Strg` + Mouse click on the title. It will do the same.

yes but i'll also mark the article as read, compared to "See on original website" arrow

**Comment by KatyPurry** [comment]
> If you use a mouse device with a scroll wheel/middle mouse button, just click with it on the title. It will open the article in a new tab on the original website. Alternative: `Strg` + Mouse click on the title. It will do the same.

Thank you for your answer. This is the method I've been using for 2 years, but I would have preferred a "native" method, i.e. using the left-click to open the link.

**Comment by Fletch-99** [comment]
I'm also interested in this. It can be very useful on mobile, since the icon "See on original website" is small for a mobile screen, so it makes sense to click on title and open the website on a new page. 

The above solution works (most of the times)
`stream.addEventListener('click', function (ev) { ev.stopPropagation(); }, true);`
but it doesn't mark the article as read.

**Comment by superkeyor** [comment]
Insert the following codes into init_stream function stream.onclick subfunction (close to the end):
You may only want to override default behavior for certain domains/rss sources. 
If needed, change the link.href.startsWith part 

```
            const link = new_active.querySelector('.item a.title');
            if (link) {
                if (link.href.startsWith('https://www.example.com') || link.href.startsWith('https://www.example.com')) {
                    if (old_active) {
                        old_active.classList.remove('active');
                        old_active.classList.remove('current'); 
                    }
                    new_active.classList.add('current');
                    mark_read(new_active, true, false);
                    window.open(link.href, '_blank');
                    return false
                }
            }
            // insert above
            toggleContent(new_active, old_active, false);
            return false;
```

**Comment by math-GH** [comment]
@superkeyor changing the core code is not suggested. It will be overwritten with the next update

**Comment by superkeyor** [comment]
> [@superkeyor](https://github.com/superkeyor) changing the core code is not suggested. It will be overwritten with the next update

Thanks for the reminder. I have been figuring out a sed command to patch the container after update, but ended up with rebuilding my own image based on the official freshrss docker image. Ideally, someone could build into the core code a setting to allow a direct url open for certain rss feed source... As for now, it works for my case. And I am sharing the solution here for others who have the same needs.

**Comment by KatyPurry** [comment]
> I'm also interested in this. It can be very useful on mobile, since the icon "See on original website" is small for a mobile screen, so it makes sense to click on title and open the website on a new page.
> 
> The above solution works (most of the times) `stream.addEventListener('click', function (ev) { ev.stopPropagation(); }, true);` but it doesn't mark the article as read.

My solution is probably not perfect but it seems more reliable : 
```JS
document.addEventListener('DOMContentLoaded', function () {
    document.getElementById('stream').addEventListener('click', function (ev) {
        let link = ev.target.closest('a');
        if (link && link.href) {
            ev.preventDefault();  // Prevent default navigation
            ev.stopImmediatePropagation();  // Stop all other event listeners (including FreshRSS)
            window.open(link.href, '_blank');  // Open link in a new tab
        }
    });

    // Also block events that might trigger article expansion
    document.getElementById('stream').addEventListener('mousedown', function (ev) {
        if (ev.target.closest('a')) {
            ev.stopImmediatePropagation();
        }
    }, true);

    document.getElementById('stream').addEventListener('mouseup', function (ev) {
        if (ev.target.closest('a')) {
            ev.stopImmediatePropagation();
        }
    }, true);
});
```
